### ISSUE TITLE
src import no longer working

### ISSUE THREAD

**Issue by stephenh** [LeDDGroup/typescript-transform-paths#132]
We'd been on 2.0.1 for awhile, and been using an import:

```
import { Icon, IconProps } from "src";
```

From a file `src/Button.tsx` with a `tsconfig` of:

```
    "paths": {
      "src/*": ["src/*"]
    },
```

Was getting correct/working output of:

```
const src_1 = require("../index");
```

However, trying out `3.2.1` we're seeing this import being left as-is:

```
const src_1 = require("src");
```

FWIW it looks like version `2.2.4` rewrites this to:

```
const src_1 = require("..");
```

Which I would believe would work. And `3.0.0` is what introduces the `require("src")` breaking change.

**Comment by nonara** [comment]
Hi Stephen! Thanks for the report. I'm going to explain what's going on, but feel free to skip to the TLDR for the solution.

Your issue actually relates to how typescript handles path mapping. In v3+, we tightened things up a bit to ensure that we're only transforming nodes which TypeScript recognizes as being a mapped path (via the compiler API) or one that is explicitly tagged with `@transform-path`. This helped improve speed and reduce unnecessary work. We also fixed some issues with how we handled implicit indexes, which is why you saw some differences in the later v2s. The current logic uses the proper handling to mirror the way nodejs require and the compiler handle implicit indexes.

Given your example, I'm not sure how `src/Button.tsx` can reference `src` (same directory) and wind up with `../index`. Maybe you meant `Button.tsx` is in a subdirectory of `src`?

In any case, the problem is that, with your current config, `src` is not recognized by TS as a mapped path. 

In all other situations, you'd end up with a typescript error saying `Cannot find module 'src' or its corresponding type declarations`. The reason that doesn't happen, however, is because it can resolve `src` _as a directory_ within your project root. So its resolution is happening outside of path mapping logic.

The short version is, the pattern `src/*` does not match `src`, as there is no slash or following characters.

## TLDR - Solution

In order to cover implicit indexes by path mapping you need two entries under paths. So for example:

```jsonc
{
  "compilerOptions": {
    "paths": {
      "src/*": [ "src/*" ],   // This covers all files within the directory
      "src": [ "src" ],       // This covers the implicit index
    }
  }
}
```

Although it may seem a little odd, this is the convention established by TS for handling that issue. 

Simply add the secondary entry for any direct paths that you want the implicit indexes to be covered, and you should be all set.

If you have any further issues, please let me know!

**Comment by stephenh** [comment]
>  Maybe you meant Button.tsx is in a subdirectory of src?

Shoot! I did mean that! :-) 

Thanks for the great response! That makes a lot of sense.
